modules/core/src/main/java/org/apache/ignite/internal/table/criteria/SqlSerializer.java

```diff
@@ -216,8 +230,13 @@ public SqlSerializer build() {
             }
 
             SqlSerializer ser = new SqlSerializer()
-                    .append("SELECT * ")
-                    .append("FROM ").append(quoteIfNeeded(tableName));
+                    .append("SELECT");
+
+            if (!nullOrBlank(indexName)) {
+                ser.append(" /*+ FORCE_INDEX(").append(indexName).append(") */");
```

Comment:
This approach allow SQL injections.

Reply:
The index name is also not something an app would be getting from the UI, so there is no way for a user to inject using that vector, but to be bulletproof added forbid characters except [a-zA-Z0-9_] in the index names passed via the Critetia API.